cadl-tests/src/test/java/com/authentication/apikey/AzureKeyCredentialPolicy.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.authentication.apikey;
+
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpPipelineNextSyncPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.HttpPipelineSyncPolicy;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Pipeline policy that uses an {@link AzureKeyCredential} to set the authorization key for a request.
+ * <p>
+ * Requests sent with this pipeline policy are required to use {@code HTTPS}. If the request isn't using {@code HTTPS}
+ * an exception will be thrown to prevent leaking the key.
+ */
+public final class AzureKeyCredentialPolicy implements HttpPipelinePolicy {
+    // AzureKeyCredentialPolicy can be a commonly used policy, use a static logger.
+    private static final ClientLogger LOGGER = new ClientLogger(com.azure.core.http.policy.AzureKeyCredentialPolicy.class);
+    private final String name;
+    private final String nameLowerCase;
+    private final AzureKeyCredential credential;
+
+    private final HttpPipelineSyncPolicy inner = new HttpPipelineSyncPolicy() {
+        @Override
+        protected void beforeSendingRequest(HttpPipelineCallContext context) {
+//            if ("http".equals(context.getHttpRequest().getUrl().getProtocol())) {
+//                throw LOGGER.logExceptionAsError(
+//                        new IllegalStateException("Key credentials require HTTPS to prevent leaking the key."));
+//            }
```

Comment:
maybe we have better way to disable this, but at present use the simplest solution.